homeassistant/components/folder/sensor.py

```diff
@@ -63,55 +67,28 @@ def setup_platform(
 class Folder(SensorEntity):
     """Representation of a folder."""
 
-    ICON = "mdi:folder"
+    _attr_device_class = SensorDeviceClass.DATA_SIZE
+    _attr_icon = "mdi:folder"
+    _attr_native_unit_of_measurement = UnitOfInformation.MEGABYTES
 
-    def __init__(self, folder_path, filter_term):
+    def __init__(self, folder_path: str, filter_term: str) -> None:
         """Initialize the data object."""
         folder_path = os.path.join(folder_path, "")  # If no trailing / add it
         self._folder_path = folder_path  # Need to check its a valid path
         self._filter_term = filter_term
-        self._number_of_files = None
-        self._size = None
-        self._name = os.path.split(os.path.split(folder_path)[0])[1]
-        self._unit_of_measurement = DATA_MEGABYTES
-        self._file_list = None
+        self._attr_name = os.path.split(os.path.split(folder_path)[0])[1]
```

Comment:
The 3 static attributes make sense to pull out from properties, but my understanding based on other components is that the `_attr_` versions are supposed to be for static definitions. So anything dynamic, like the rest of the changes, would still use the custom property override to indicate that they are dynamically computed.

Reply:
No, we prefer instance attributes. In some cases though, eg coordinator entities with a single state attribute to update, a property for the native value is cleaner.